src/ssh-util/src/tunnel.rs

```diff
@@ -123,6 +126,26 @@ impl SshTunnelConfig {
             };
         };
 
+        // Ensure the tunnel is actually established.
+        attempts = 0;
+        loop {
+            if attempts > 50 {
+                bail!("SSH tunnel could not actually be connected to")
+            } else {
+                attempts += 1;
+            }
+
+            if let Ok(stream) =
+                tokio::net::TcpStream::connect((Ipv4Addr::LOCALHOST, local_port)).await
+            {
+                // Ensure the socket is writable, just in case.
+                stream.writable().await?;
+                break;
+            }
+            // This is rarely hit, but could slow down certain ddl statements if it is.
+            tokio::time::sleep(std::time::Duration::from_millis(50)).await;
+        }
```

Comment:
Did this actually wind up being necessary? If you aren’t able to repro the ECONNREFUSED without this (but with the other changes), I’m inclined to leave it out. Otherwise this sort of stuff becomes software superstition—we’ll be too afraid to ever remove it out of fear that one customer somewhere needs it. Whereas it’s very easy to go in the opposite direction and add it if proven necessary.

Reply:
It didnt! Or at least, it didn't when I tested removing it, before my macbook's ssh started not working at all! Removed in force-pushed update!